socorro/unittest/external/es/test_supersearch.py

```diff
@@ -734,6 +734,87 @@ def test_get_with_facets(self):
         )
 
     @minimum_es_version('1.0')
+    def test_get_with_cardinality(self):
+        self.index_crash({
+            'signature': 'js::break_your_browser',
+            'product': 'WaterWolf',
+            'os_name': 'Windows NT',
+            'date_processed': self.now,
+        })
+        self.index_crash({
+            'signature': 'js::break_your_browser',
+            'product': 'WaterWolf',
+            'os_name': 'Linux',
+            'date_processed': self.now,
+        })
+        self.index_crash({
+            'signature': 'js::break_your_browser',
+            'product': 'NightTrain',
+            'os_name': 'Linux',
+            'date_processed': self.now,
+        })
+        self.index_crash({
+            'signature': 'foo(bar)',
+            'product': 'EarthRacoon',
+            'os_name': 'Linux',
+            'date_processed': self.now,
+        })
+
+        # Index a lot of distinct values.
+        number_of_crashes = 51
+        processed_crash = {
+            'version': '10.%s',
+            'date_processed': self.now,
+        }
+        self.index_many_crashes(
+            number_of_crashes,
+            processed_crash,
+            loop_field='version',
+        )
+        # Note: index_many_crashes does the index refreshing.
+
+        # Test a simple cardinality.
+        kwargs = {
+            '_facets': ['_cardinality.platform']
+        }
+        res = self.api.get(**kwargs)
+
+        ok_('facets' in res)
+        ok_('cardinality_platform' in res['facets'])
+        expected_value = {'value': 2}
+        eq_(res['facets']['cardinality_platform'], expected_value)
+
+        # Test more distinct values.
+        kwargs = {
+            '_facets': ['_cardinality.version']
+        }
+        res = self.api.get(**kwargs)
+
+        ok_('facets' in res)
+        ok_('cardinality_version' in res['facets'])
+        expected_value = {'value': 51}
+        eq_(res['facets']['cardinality_version'], expected_value)
+
+        # Test as a level 2 aggregation.
+        kwargs = {
+            '_aggs.signature': ['_cardinality.platform']
+        }
+        res = self.api.get(**kwargs)
+
+        ok_('facets' in res)
+        ok_('signature' in res['facets'])
```

Comment:
grr. Ignore this comment. I see now below.

Nit: remove this line break so that the `for facet in res['facets']['signature']:` stuff comes immediately after `ok_('signature' in res['facets'])`